web-admin/src/features/scheduled-reports/query-mapper.ts

```diff
@@ -0,0 +1,277 @@
+      metricsViewName = (req as V1MetricsViewToplistRequest).metricsViewName;
+      queryMapper = mapMetricsViewToplistRequest;
+      break;
+    case "MetricsViewRows":
+      metricsViewName = (req as V1MetricsViewRowsRequest).metricsViewName;
+      queryMapper = mapMetricsViewRowsRequest;
+      break;
+    case "MetricsViewTimeSeries":
+      metricsViewName = (req as V1MetricsViewTimeSeriesRequest).metricsViewName;
+      queryMapper = mapMetricsViewTimeSeriesRequest;
+      break;
+    case "MetricsViewComparison":
+      metricsViewName = (req as V1MetricsViewComparisonRequest).metricsViewName;
+      queryMapper = mapMetricsViewComparisonRequest;
+      break;
+  }
+
+  if (!metricsViewName) {
+    // error state
+    return readable({
+      ready: true,
+      error:
+        "Failed to find metrics view name. Please check the format of the report.",
+    });
+  }
+
+  return derived(
+    [
+      useMetaQuery(get(runtime).instanceId, metricsViewName),
+      // TODO: handle non-timestamp dashboards
+      createQueryServiceMetricsViewTimeRange(
+        get(runtime).instanceId,
+        metricsViewName,
+        {},
+      ),
+    ],
+    ([metricsViewResource, timeRangeSummary]) => {
+      if (!metricsViewResource.data || !timeRangeSummary.data)
+        return {
+          ready: false,
+        };
+
+      if (metricsViewResource.error || timeRangeSummary.error) {
+        // error state
+        return {
+          ready: true,
+          error:
+            metricsViewResource.error?.message ??
+            timeRangeSummary.error?.message,
+        };
+      }
+
+      initLocalUserPreferenceStore(metricsViewName);
+      const dashboard = getDefaultMetricsExplorerEntity(
+        metricsViewName,
+        metricsViewResource.data,
+        timeRangeSummary.data,
+      );
+      queryMapper({
+        dashboard,
+        req,
+        metricsView: metricsViewResource.data,
+        timeRangeSummary: timeRangeSummary.data.timeRangeSummary,
+        executionTime,
+      });
+      console.log(dashboard);
+      return {
+        ready: true,
+        state: getProtoFromDashboardState(dashboard),
+        metricsView: metricsViewName,
+      };
+    },
+  );
+}
+
+/**
+ * This method corrects the underscore naming to camel case.
+ * This is the drawback of storing the request object as is.
+ */
+function correctRequest(req: Record<string, any>): Record<string, any> {
+  const newReq: Record<string, any> = {};
+
+  for (const key in req) {
+    const newKey = key.replace(/_(\w)/g, (_, c: string) => c.toUpperCase());
+    const val = req[key];
+    if (val && typeof val === "object" && !("length" in val)) {
+      newReq[newKey] = correctRequest(val);
+    } else {
+      newReq[newKey] = val;
+    }
+  }
+
+  return newReq;
+}
+
+function mapMetricsViewAggregationRequest({
+  req,
+  dashboard,
+}: QueryMapperArgs<V1MetricsViewAggregationRequest>) {
+  if (req.where) dashboard.whereFilter = req.where;
+  // TODO
```

Comment:
Mind clarifying that these TODOs should be completed when we offer scheduled reports for the given query?

Reply:
UI only adds MetricsViewComparison right now. Need to clarify if we need to add support for others in the 1st cut. Lemme double check with product before merging this PR.

Reply:
Probably not needed for the first cut because we don't yet mention Report YAML in our docs anywhere.

My point was that the TODOs confused me until I realized that we don't yet offer scheduled reports for these queries. A note like 'TODO: Scheduled reports not yet available for this query' could help clarify.